src/beta_inc.jl

```diff
@@ -0,0 +1,886 @@
+    end
+    z = a*lnx + b*lny
+    if a0 < 1.0
+        b0 = max(a,b)
+        if b0 >= 8.0
+            u = SpecialFunctions.loggamma1p(a0) + loggammadiv(a0,b0)
+            return a0*(case ? esum(mu, z-u) : exp(z-u))
+        elseif b0 > 1.0
+            u = SpecialFunctions.loggamma1p(a0)
+            n = trunc(Int,b0 - 1.0)
+            if n >= 1
+                c = 1.0
+                for i = 1:n
+                    b0 -= 1.0
+                    c *= (b0/(a0+b0))
+                end
+                u += log(c)
+            end
+            z -= u
+            b0 -= 1.0
+            apb = a0 + b0
+            if apb > 1.0
+                u = a0 + b0 - 1.0
+                t = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+            else
+                t = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+            end
+            return a0*(case ? esum(mu,z) : exp(z))*(1.0 + SpecialFunctions.rgamma1pm1(b0))/t
+        end
+    else
+        z -= logbeta(a,b)
+        ans = case ? esum(mu,z) : exp(z)
+        return ans
+    end
+
+   
+    
+    if ans == 0.0
+        return 0.0
+    end
+    apb = a + b
+    if apb > 1.0
+        u = a + b - 1.0
+        z = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+    else
+        z = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+    end
+    c = (1.0 + SpecialFunctions.rgamma1pm1(a))*(1.0 + SpecialFunctions.rgamma1pm1(b))/z
+    return ans*(a0*c)/(1.0 + a0/b0)    
+end  
+
+"""
+    beta_inc_cont_fraction(a,b,x,y,lambda,epps)
+
+Compute ``I_{x}(a,b)`` using continued fraction expansion when a,b > 1.
+It is assumed that ``\\lambda = (a+b)*y - b``
+DLMF : https://dlmf.nist.gov/8.17#E22
+BFRAC(A,B,X,Y,LAMBDA,EPS)
+"""
+function beta_inc_cont_fraction(a::Float64, b::Float64, x::Float64, y::Float64, lambda::Float64, epps::Float64)
+    ans = brcmp1(0.0,a,b,x,y,false)
+    if ans == 0.0
+        return 0.0
+    end
+    c = 1.0 + lambda
+    c0 = b/a
+    c1 = 1.0 + 1.0/a
+    yp1 = y + 1.0
+
+    n = 0.0
+    p = 1.0
+    s = a + 1.0
+    an = 0.0
+    bn = 1.0
+    anp1 = 1.0
+    bnp1 = c/c1
+    r = c1/c
+    #CONT FRACTION
+
+    while true
+     n += 1.0
+     t = n/a
+     w = n*(b - n)*x
+     e = a/s
+     alpha = (p*(p+c0)*e*e)*(w*x)
+     e = (1.0 + t)/(c1 + 2*t)
+     beta = n + w/s +e*(c + n*yp1)
+     p = 1.0 + t
+     s += 2.0
+
+     #update an, bn, anp1, bnp1
+     t = alpha*an  + beta*anp1
+     an = anp1
+     anp1 = t
+     t = alpha*bn + beta*bnp1
+     bn = bnp1
+     bnp1 = t
+
+     r0 = r
+     r = anp1/bnp1
+     if abs(r - r0) <= epps*r
+        break
+     end
+     #rescale
+     an /= bnp1
+     bn /= bnp1
+     anp1 = r
+     bnp1 = 1.0
+    end
+    return ans*r
+end
+
+"""
+    beta_inc_asymp_exp(a,b,lambda,epps)
+
+Compute ``I_{x}(a,b)`` using asymptotic expansion for a,b >= 15.
+It is assumed that ``\\lambda = (a+b)*y - b``
+BASYM(A,B,LAMBDA,EPS)
+"""
+function beta_inc_asymp_exp(a::Float64, b::Float64, lambda::Float64, epps::Float64)
+    a0 = b0 = c = d = zeros(22)
```

Comment:
I think this is the issue: this is assigning the _same_ array to all variables (so mutating one of them mutates all of them)
```suggestion
    a0 = zeros(22)
    b0 = zeros(22)
    c = zeros(22)
    d = zeros(22)
```

Reply:
OH NO!!!!
That was sad. Too much accustomed to using these in C++. 
Thanks a lot anyways.